onnx/reference/ops/op_quantize_linear.py

```diff
@@ -213,3 +217,24 @@ def _run(
             output_dtype=output_dtype,
             precision=precision,
         )
+
+
+class QuantizeLinear_25(_CommonQuantizeLinear):
```

Comment:
Just commenting on the basic underlying implementation (not necessary to fix in this PR): this seems a bit flawed if the base implementation supports all types. In a reference implementation, it should fail if inputs are not of type allowed by the spec. So, opset 24 implementation should fail for INT2, but opset 25 implementation should not.